### PR TITLE
(6.1) Refactor robotest configuration - keep upgrade_from releases current.

### PR DESCRIPTION
**Pull request by wadells**
## Description
Port robotest config refactoring from #1837 

This port introduces (or rather carries forward) some additions not in master (#1837), 7.0 (#1876), or 6.3 (#1877):

Specifically 6.1 and 5.5 have a list of custom 5.5 versions to test upgrade from, and we see that start to crop up in this PR.  I also edited these from "ubuntu:16"  (previously on 6.1.x) to "centos:7" as I believe this is closer to what we see these versions running on in the field.

## Type of change
* Internal change (not necessarily a bug fix or a new feature)

## Linked tickets and other PRs
* Ports #1837 

## TODOs
- [x] Self-review the change
- [x] Audit automated test run for correctness
- [x] Address review feedback

## Testing done
See  #1837 for the original testing. For 6.1: the PR Jenkins run should sufficiently exercise the 6.1 PR build relevant changes. We don't care about testing the nightly changes because we don't run nighties on 6.1 so although I made an minor effort to keep this file branch relevant mistakes are moot.

### CODE REVIEW

**Review by r0mant — COMMENTED** [review]
Robotest branch is failing with

```
03:53:52 [robotest] [31m[ERROR]: unknown long flag '--hub'
03:53:52 [robotest] [0mmake[1]: *** [robotest-run-suite] Error 255
```

Might be related to the change in the version of `tele` being used? The `--hub` flag might have been added to `tele login` command after 6.1.

**Comment by wadells** [comment]
Fixed the `tele` incompatiblity as well as removed problematic `centos:8` from the config.

Here is the diff between the new config (at 5985086) and the old config (pre-this-pr):

```
walt@work:~/git/gravity$ diff pr.before pr.after                             
2,3c2,3
< install={"installer_url":"/installer/opscenter.tar","nodes":1,"flavor":"standalone","role":"node","os":"ubuntu:16","ops_advertise_addr":"example.com:443"}
< resize={"to":3,"flavor":"one","nodes":1,"role":"node","state_dir":"/var/lib/telekube","os":"ubuntu:16","storage_driver":"overlay2"}
---
> install={"installer_url":"/installer/opscenter.tar","nodes":1,"flavor":"standalone","role":"node","os":"ubuntu:18","ops_advertise_addr":"example.com:443"}
> resize={"to":3,"flavor":"one","nodes":1,"role":"node","state_dir":"/var/lib/telekube","os":"ubuntu:18","storage_driver":"overlay2"}
5,12c5,12
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:16","storage_driver":"overlay2","from":"/telekube_5.5.0.tar"}
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:16","storage_driver":"overlay2","from":"/telekube_5.5.28.tar"}
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:16","storage_driver":"overlay2","from":"/telekube_5.5.36.tar"}
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:16","storage_driver":"overlay2","from":"/telekube_5.5.38.tar"}
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:16","storage_driver":"overlay2","from":"/telekube_5.5.40.tar"}
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:16","storage_driver":"overlay2","from":"/telekube_5.5.41.tar"}
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:16","storage_driver":"overlay2","from":"/telekube_6.1.0.tar"}
< upgrade={"flavor":"three","nodes":3,"role":"node","service_uid":997,"service_gid":994,"os":"ubuntu:16","storage_driver":"overlay2","from":"/telekube_6.1.25.tar"}
---
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"centos:7","from":"/telekube_5.5.0.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"centos:7","from":"/telekube_5.5.28.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"centos:7","from":"/telekube_5.5.36.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"centos:7","from":"/telekube_5.5.38.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"centos:7","from":"/telekube_5.5.40.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"redhat:7","from":"/telekube_5.5.49.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"ubuntu:16","from":"/telekube_6.1.0.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
> upgrade={"flavor":"three","nodes":3,"role":"node","os":"ubuntu:18","from":"/telekube_6.1.31.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
```

Of note:

* install and resize are moved from `ubuntu:16` to `ubuntu:18`.  This wasn't intentional, but will save merge conflicts so "why not?" :shrug:.
* The specific hardcoded 5.5.x versions are changed to `centos:7` as I believe this is closer to what we commonly see these running on in the field.
* We move from testing `ubuntu:16` and `redhat:7` to `ubuntu:16`, `ubuntu:18`, `redhat:7` and `centos:7`.  Debian is notably missing, not for any particular reason other than it was assigned 5.6, and those are commented out.  I can add it anyone thinks that is important.
* And of course, we get the most current patch release automatically!

File: build.assets/robotest/nightly_config.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  done
+  suite+=$(build_ops_install_suite)
+  echo -n $suite
+}
+
+SUITE="$(build_install_suite)"
+SUITE+=" $(build_resize_suite)"
+SUITE+=" $(build_upgrade_suite)"
+
+# robotest wants the suite space seperated, but it is easier to eyeball when newline seperated
```
Comment (a-palchikov) [inline build.assets/robotest/nightly_config.sh:104]: nit: sep**a**rated (2x)

**Review by a-palchikov — APPROVED** [review]

File: build.assets/robotest/nightly_config.sh (hunk 1 above)
Comment (wadells) [reply build.assets/robotest/nightly_config.sh]: Fixed in ee55e37.

**Pull request merged**